Wrappers/Python/cil/optimisation/operators/GradientOperator.py

```diff
@@ -263,12 +280,39 @@ class Gradient_C(LinearOperator):
 
     def __init__(self, gm_domain, gm_range=None, bnd_cond = NEUMANN, **kwargs):
 
+        # Number of threads
         self.num_threads = kwargs.get('num_threads',NUM_THREADS)
+        
+        # Split gradients, e.g., space and channels
         self.split = kwargs.get('split',False)
+        
         self.gm_domain = gm_domain
         self.gm_range = gm_range
-        self.ndim = self.gm_domain.length
-                
+        
+        # Consider pseudo 3D geometries with one slice, e.g., (1,155,155)
+        self.tmp_dom_shape = list(gm_domain.shape)               
+        self.ind = list(range(len(self.tmp_dom_shape)))
+        self.is2D = False
+        
+        # List of voxel sizes
+        tmp_gm_domain_spacing = list(self.gm_domain.spacing)
+
+        for i, size in enumerate(self.tmp_dom_shape):
+            if size==1:
+                self.tmp_dom_shape.remove(size)
```

Comment:
I'm worried that removing from the list we are iterating might get us into troubles? @gfardell do you use this trick? 
I'd personally construct a new list appending if `size != 1` rather than remove.

Reply:
fixed